main.tf

```diff
@@ -61,8 +61,31 @@ resource "aws_lambda_permission" "sleuth_lambda_permission" {
 #
 # IAM
 #
-
 data "aws_iam_policy_document" "task_role_policy_doc" {
+  count = var.enable_sns_topic ? 0 : 1
+  # Allow to list and disable keys
+  statement {
+    actions = [
+      "iam:UpdateAccessKey",
+      "iam:ListAccessKeys",
+      "iam:ListUserTags",
+    ]
+
+    resources = ["arn:aws:iam::*:user/*"]
+  }
+
+  # Allow to list and disable keys
+  statement {
+    actions = [
+      "iam:ListUsers",
+    ]
+
+    resources = ["*"]
+  }
+}
```

Comment:
Don't you need that regardless of how you're sending the notification?

Reply:
i think you got it but, just in case, creating two different policies where one has the sns topic created and the other doesn't. or can it be simplified?

Reply:
I was just thinking about having three policies -- one with the stuff you need to make it work at all, one with the stuff for slack, one with the stuff for SNS. That might be *more* complicated though, which isn't necessarily great.

Reply:
i don't think i need a separate policy around slack (at least the direct one through the webhook). am i wrong?

Reply:
I wasn't sure it if was an "in addition to" or an "instead of". :) I will defer to your knowledge in this case. :)